sycl/include/CL/sycl/sycl_span.hpp

```diff
@@ -0,0 +1,619 @@
+    constexpr span() noexcept;
+    constexpr explicit(Extent != dynamic_extent) span(pointer ptr, size_type
+count); constexpr explicit(Extent != dynamic_extent) span(pointer firstElem,
+pointer lastElem); template <size_t N> constexpr span(element_type (&arr)[N])
+noexcept; template <size_t N> constexpr span(array<value_type, N>& arr)
+noexcept; template <size_t N> constexpr span(const array<value_type, N>& arr)
+noexcept; template <class Container> constexpr explicit(Extent !=
+dynamic_extent) span(Container& cont); template <class Container> constexpr
+explicit(Extent != dynamic_extent) span(const Container& cont); constexpr
+span(const span& other) noexcept = default; template <class OtherElementType,
+size_t OtherExtent> constexpr explicit(Extent != dynamic_extent) span(const
+span<OtherElementType, OtherExtent>& s) noexcept; ~span() noexcept = default;
+    constexpr span& operator=(const span& other) noexcept = default;
+
+    // [span.sub], span subviews
+    template <size_t Count>
+        constexpr span<element_type, Count> first() const;
+    template <size_t Count>
+        constexpr span<element_type, Count> last() const;
+    template <size_t Offset, size_t Count = dynamic_extent>
+        constexpr span<element_type, see below> subspan() const;
+
+    constexpr span<element_type, dynamic_extent> first(size_type count) const;
+    constexpr span<element_type, dynamic_extent> last(size_type count) const;
+    constexpr span<element_type, dynamic_extent> subspan(size_type offset,
+size_type count = dynamic_extent) const;
+
+    // [span.obs], span observers
+    constexpr size_type size() const noexcept;
+    constexpr size_type size_bytes() const noexcept;
+    constexpr bool empty() const noexcept;
+
+    // [span.elem], span element access
+    constexpr reference operator[](size_type idx) const;
+    constexpr reference front() const;
+    constexpr reference back() const;
+    constexpr pointer data() const noexcept;
+
+    // [span.iterators], span iterator support
+    constexpr iterator begin() const noexcept;
+    constexpr iterator end() const noexcept;
+    constexpr reverse_iterator rbegin() const noexcept;
+    constexpr reverse_iterator rend() const noexcept;
+
+private:
+    pointer data_;    // exposition only
+    size_type size_;  // exposition only
+};
+
+template<class T, size_t N>
+    span(T (&)[N]) -> span<T, N>;
+
+template<class T, size_t N>
+    span(array<T, N>&) -> span<T, N>;
+
+template<class T, size_t N>
+    span(const array<T, N>&) -> span<const T, N>;
+
+template<class Container>
+    span(Container&) -> span<typename Container::value_type>;
+
+template<class Container>
+    span(const Container&) -> span<const typename Container::value_type>;
+
+} // namespace std
+
+*/
+
+#include <array>       // for array
+#include <cstddef>     // for byte
+#include <iterator>    // for iterators
+#include <type_traits> // for remove_cv, etc
+
+#define _SYCLSPAN_TEMPLATE_VIS
```

Comment:
Could you please clarify what this macro is for and why it defined to nothing?

Reply:
The `_SYCLSPAN_` macros (only 2 or three) replace the `_LIBCPP_` originals, to avoid name conflicts. There is a comment at the top of the file mentioning this.   Even though it's defined to nothing in our case, I was leaving the macros in.  Our `_SYCLSPAN_TEMPLATE_VIS` macro stands in place of the original `_LIBCPP_TEMPLATE_VIS` .

Reply:
Characters are cheap nowadays. Add this as a comment so no one will ask again. :-)

Reply:
I thought you changed all the `_SYCLSPAN...` with `_SYCL_SPAN...` but it it is not the case...

Reply:
done